### ISSUE TITLE
reset Password gives error for old users (zk 1.4.6 RC1)

### ISSUE THREAD

**Issue by rallek** [zikula/core#3452]
| Q                 | A
| ----------------- | ---
| Zikula Version    | 1.4.6 ci 111
| PHP Version       | 5.5

### Expected behavior
Password reset works propperly

### Actual behavior
````
[1] Symfony\Component\Debug\Exception\FatalErrorException: Error: Call to a member function setPass() on a non-object
    at n/a
        in /www/htdocs/w0158c4d/Zikula_Core-1.4.6/system/ZAuthModule/Controller/AccountController.php line 240
````
no further stack trace is showing

### Steps to reproduce
We have user from time before zk native authorization. If they request a password recovery they do get an email with the rest link. Following the link they are able to enter a new password. But after saving they do get the above error message.

If I look into database the user do have an entry in the table users_verifychg.

The users are imported from an old 1.2.7 installation. They are shown correctly in the UsersModule. Some weeks of testing ago I was able to pass the complete procedure with an user from this time.

A user from newer time who is already registered via zk native authorization is able to reset his password propperly.

**Comment by craigh** [comment]
> Some weeks of testing ago I was able to pass the complete procedure with an user from this time

this is impossible. there has never been code to make this work.

**Comment by craigh** [comment]
hmmm. maybe I'm wrong. maybe refs [#3227](https://github.com/zikula/core/pull/3277)

**Comment by craigh** [comment]
refs #3274

**Comment by craigh** [comment]
I don't think that it ever worked. Please test my fix to see if it fixes your issue.

**Comment by rallek** [comment]
The error has gone. Now I do get this failure message from zikula:
````
Your request could not be processed. Unexpected extraction results in lost password verification helper.
````

**Comment by Guite** [comment]
This was caused by a special case which is now fixed in 221cc167e334510f1de890155662ea8b154e8170
